x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfig.java

```diff
@@ -69,6 +69,12 @@ public class AnalysisConfig implements ToXContentObject, Writeable {
     public static final String ML_CATEGORY_FIELD = "mlcategory";
     public static final Set<String> AUTO_CREATED_FIELDS = new HashSet<>(Collections.singletonList(ML_CATEGORY_FIELD));
 
+    // Since the C++ backend truncates the categorization field at length 1000, adding an ellipsis on truncation, it makes no sense to
+    // send potentially very long strings to it. For the backend logic still to work we need to send more than that, hence we truncate
+    // at length 1001.
+    //
+    // Also, because we do the tokenization on the Java side now the tokens will still be sent correctly (separately) to the C++ backend
```

Comment:
I think the time has come to delete the constant and the 6 places where it's used: https://github.com/elastic/elasticsearch/blob/9e530e163cf7ee89fc7133b86b02a465ab9c6413/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java#L487

Then this comment really will be true always.